### PR TITLE
FIX: support latin1 in channels in nihon files

### LINKED ISSUE #10428: support latin1 encoding for channels in read_raw_nihon
When opening Nihon-Kohden files with `read_raw_nihon` I get the error

```
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc4 in position 1070: invalid continuation byte
```
raised in [this line](https://github.com/mne-tools/mne-python/blob/1b14d458cbcd391a6907032b809ec37410b60592/mne/io/nihon/nihon.py#L111).

I guess it is similar to [this issue](https://github.com/mne-tools/mne-python/issues/9267) which was already solved [here](https://github.com/mne-tools/mne-python/pull/9384), as replacing the above mentioned line with 
```
with open(e_fname, 'r', encoding='latin1') as fid:
```

fixed the issue.

Maybe we could add a similar [loop over different encodings](https://github.com/mne-tools/mne-python/pull/9384/files#diff-800fa074fc54d9bab7e81ae7d9f3239321fdf53d127bf6bbea3935a3095586e5R252) in [_read_21e_file](https://github.com/mne-tools/mne-python/blob/1b14d458cbcd391a6907032b809ec37410b60592/mne/io/nihon/nihon.py#L102)?

### PR DESCRIPTION
**Pull request by MatthiasEb**
Fixes #10428 similar to [this PR](https://github.com/mne-tools/mne-python/pull/9384)

### CODE REVIEW

**Comment by agramfort** [comment]
you need to check flake8 issues @MatthiasEb 

you can type

make flake

in the mne-python folder to check locally.

**Comment by MatthiasEb** [comment]
Sorry, I don't know how to fix these failed checks that succeeded before, I barely changed anything.

**Review by larsoner — COMMENTED** [review]
The CI failure was spurious, restart fixed it.

@MatthiasEb can you add an entry to `doc/changes/latest.inc` in the BUG section? I think that's the last thing we need, then we can merge!

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -130,6 +130,7 @@ Enhancements
 
 Bugs
 ~~~~
+- Fix bug with :func:`mne.io.read_raw_nihon` where latin-1 channels could not be read (:gh:`10429` :newcontrib:`Matthias Eberlein`_)
```
Comment (larsoner) [inline doc/changes/latest.inc:133]: You'll need to add a link to `doc/changes/names.inc` as well. If you put it in the middle somewhere (as opposed to the very bottom) you're less likely to hit a confilct

File: doc/changes/latest.inc [hunk 2]
```diff
@@ -130,6 +130,7 @@ Enhancements
 
 Bugs
 ~~~~
+- Fix bug with :func:`mne.io.read_raw_nihon` where latin-1 channels could not be read (:gh:`10429` :newcontrib:`Matthias Eberlein`)
```
Comment (cbrnr) [inline doc/changes/latest.inc:133]:
```suggestion
- Fix bug with :func:`mne.io.read_raw_nihon` where latin-1 channels could not be read (:gh:`10429` by :newcontrib:`Matthias Eberlein`)
```

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
thx @MatthiasEb !

**Comment by MatthiasEb** [comment]
No worries, it's been a pleasure. Thanks for your support.

**Pull request merged**